### PR TITLE
Payments sync fix transaction block

### PR DESCRIPTION
**Pull request by ashwin1111**

### CODE REVIEW

File: apps/netsuite/tasks.py [hunk 1]
```diff
@@ -871,10 +871,6 @@ def create_netsuite_payment_objects(netsuite_objects, object_type, workspace_id)
                             'expense_group': netsuite_object.expense_group,
                         }
                     )
-        else:
```
Comment (ashwin1111) [inline apps/netsuite/tasks.py:874]: this is kind of not needed, we already do this while creating vendor payments

File: apps/netsuite/tasks.py [hunk 2]
```diff
@@ -891,120 +887,116 @@ def process_vendor_payment(entity_object, workspace_id, object_type):
...
     )
-    try:
-        vendor_payment_object = VendorPayment.create_vendor_payment(
-            workspace_id, entity_object
-        )
-
-        vendor_payment_lineitems = VendorPaymentLineitem.create_vendor_payment_lineitems(
-            entity_object['line'], vendor_payment_object
-        )
+    with transaction.atomic():
```
Comment (ashwin1111) [inline apps/netsuite/tasks.py:890]: added transaction block

File: apps/netsuite/tasks.py [hunk 3]
```diff
@@ -891,120 +887,116 @@ def process_vendor_payment(entity_object, workspace_id, object_type):
...
+            detail = exception.response
+            task_log.status = 'FAILED'
+            task_log.detail = detail
 
-        task_log.save()
+            task_log.save()
 
 
 def create_vendor_payment(workspace_id):
-    try:
```
Comment (ashwin1111) [inline apps/netsuite/tasks.py:971]: removed try except in this function, not needed

File: apps/netsuite/tasks.py [hunk 4]
```diff
@@ -891,120 +887,116 @@ def process_vendor_payment(entity_object, workspace_id, object_type):
...
-        expense_reports = ExpenseReport.objects.filter(
-            payment_synced=False, expense_group__workspace_id=workspace_id,
-            expense_group__fund_source='PERSONAL', expense_group__exported_at__isnull=False
-        ).all()
+    expense_reports = ExpenseReport.objects.filter(
+        payment_synced=False, expense_group__workspace_id=workspace_id,
+        expense_group__fund_source='PERSONAL', expense_group__exported_at__isnull=False
+    ).all()
 
-        with transaction.atomic():
```
Comment (ashwin1111) [inline apps/netsuite/tasks.py:986]: removed transaction block

File: apps/netsuite/tasks.py (hunk 2 above)
Comment (ashwin1111) [reply apps/netsuite/tasks.py:890]: nothing else changed here

**Review by Shwetabhk — APPROVED** [review]

**Pull request merged**